server/graphql/v2/interface/HasMembers.js

```diff
@@ -38,6 +45,14 @@ export const HasMembersFields = {
         };
       }
 
+      if (args.email) {
+        if (!req.remoteUser.isAdminOfCollective(collective)) {
+          throw new BadRequest('Only admins can lookup for members using the "email" argument');
+        } else {
+          collectiveInclude.push({ association: 'user', required: true, where: { email: args.email.toLowerCase() } });
```

Comment:
I think we should filter out incognito collectives.

Reply:
I was wondering about this one, but it seems that historically collective admins have access to incognito user's emails: https://github.com/opencollective/opencollective-api/blob/661638353b7287fd5c590393592ec5a6946244a5/server/graphql/v1/types.js#L368.

Keeping things this way is actually interesting for Athens: they should be able to verify if someone contributed - regardless of whether it was made as an incognito contribution or not.

Reply:
In our case, users want us, the collective they are sponsoring to have their email address. This makes sense if you think about Substack's case about helping creators (collectives?) maintain a direct relationship with their readers.